deepchecks/vision/utils/detection_encoders.py

```diff
@@ -0,0 +1,172 @@
+        If a string, it must be one of the following:
+            - 'xyxy' - x1, y1, x2, y2 represent the upper left and lower right corners of the bounding box.
+            - 'xywh' - x1, y1, w, h represent the upper left corner of the bounding box and its width and height.
+            - 'cxcywh' - x, y, w, h represent the center of the bounding box and its width and height.
+            - 'xyxyn' - x1, y1, x2, y2, n represent the upper left and lower right corners of the bounding box,
+                        normalized to the image dimensions.
+            - 'xywhn' - x1, y1, w, h, represent the upper left corner of the bounding box and its width and height,
+                        normalized by the image dimensions.
+            - 'cxcywhn' - (YOLO format) x, y, w, h, represent the center of the bounding box and its width and height,
+                          normalized by the image dimensions.
+        In addition, the label shape should be a list of length N containing tensors of shape (M, 5), where N is the
+        number of samples, M is the number of bounding boxes, and each bounding box is represented by 5 values:
+        (class_id, 4 coordinates in the format specified by the `label_encoder` parameter).
+
+        If a function, it must follow the signature:
+        Function that takes in a batch of labels and returns the encoded labels in the following format:
+        List of length N containing tensors of shape (M, 5), where N is the number of samples,
+        M is the number of bounding boxes in the sample and each bounding box is represented by 5 values: (class_id,
+        x, y, w, h). x and y are the coordinates (in pixels) of the upper left corner of the bounding box, w and h are
+        the width and height of the bounding box (in pixels) and class_id is the class id.
+
+    """
+
+    label_encoder: Union[str, Callable]
+
+    def __init__(self, label_encoder: Union[str, Callable]):
+        super().__init__(label_encoder)
+        self.label_encoder = label_encoder
+
+    def __call__(self, *args, **kwargs):
+        """Call the encoder."""
+        if isinstance(self.label_encoder, Callable):
+            return self.label_encoder(*args, **kwargs)
+        elif isinstance(self.label_encoder, str):
+            pass
+
+    def get_samples_per_class(self, data_loader: DataLoader):
+        """
+        Get the number of samples per class.
+
+        Parameters
+        ----------
+        data_loader : DataLoader
+            DataLoader to get the samples per class from.
+
+        Returns
+        -------
+        Counter
+            Counter of the number of samples per class.
+        """
+        counter = Counter()
+        for _ in range(len(data_loader)):
+            list_of_arrays = self(next(iter(data_loader))[1])
+            class_list = sum([arr.reshape((-1, 5))[:, 0].tolist() for arr in list_of_arrays], [])
+            counter.update(class_list)
+
+        return counter
```

Comment:
There are relevant functions in TrainTestLabelDrift that should be moved here, IMO - calculating bbox areas and counting bboxes in images

Reply:
I am not sure. Perhaps we should calculate it once using the COCO JSON format. @JKL98ISR is still figuring out how to make this work.